kernel: "New timer subsystem" uses oversimplified, common names, leading to debugging confusion and compilation clashes

From https://github.com/zephyrproject-rtos/zephyr/pull/10303#issuecomment-430638826 : 

~~~
/home/buildslave/workspace/lite-aeolus-micropython/zephyr/kernel/timeout.c:47:13: error: conflicting types for 'remove'
12:22:03  static void remove(struct _timeout *t)
12:22:03              ^~~~~~
12:22:03 In file included from /home/buildslave/workspace/lite-aeolus-micropython/zephyr/ext/hal/st/stm32cube/stm32f4xx/drivers/include/stm32f4xx_hal_def.h:48:0,
12:22:03                  from /home/buildslave/workspace/lite-aeolus-micropython/zephyr/ext/hal/st/stm32cube/stm32f4xx/drivers/include/stm32f4xx_hal_rcc.h:45,
12:22:03                  from /home/buildslave/workspace/lite-aeolus-micropython/zephyr/ext/hal/st/stm32cube/stm32f4xx/drivers/include/stm32f4xx_hal_conf.h:259,
12:22:03                  from /home/buildslave/workspace/lite-aeolus-micropython/zephyr/ext/hal/st/stm32cube/stm32f4xx/drivers/include/stm32f4xx_hal.h:46,
12:22:03                  from /home/buildslave/workspace/lite-aeolus-micropython/zephyr/ext/hal/st/stm32cube/stm32f4xx/soc/stm32f4xx.h:250,
12:22:03                  from /home/buildslave/workspace/lite-aeolus-micropython/zephyr/soc/arm/st_stm32/stm32f4/soc.h:27,
12:22:03                  from /home/buildslave/workspace/lite-aeolus-micropython/zephyr/include/arch/arm/cortex_m/cmsis.h:21,
12:22:03                  from /home/buildslave/workspace/lite-aeolus-micropython/zephyr/arch/arm/include/cortex_m/stack.h:27,
12:22:03                  from /home/buildslave/workspace/lite-aeolus-micropython/zephyr/arch/arm/include/kernel_arch_data.h:52,
12:22:03                  from /home/buildslave/workspace/lite-aeolus-micropython/zephyr/kernel/include/kernel_structs.h:65,
12:22:03                  from /home/buildslave/workspace/lite-aeolus-micropython/zephyr/kernel/include/ksched.h:10,
12:22:03                  from /home/buildslave/workspace/lite-aeolus-micropython/zephyr/kernel/timeout.c:10:
12:22:03 /home/buildslave/srv/toolchain/zephyr-sdk-0.9.3/sysroots/armv5-zephyr-eabi/usr/include/stdio.h:227:5: note: previous declaration of 'remove' was here
12:22:03  int _EXFUN(remove, (const char *));
12:22:03      ^~~~~~
~~~

--- a/kernel/timeout.c
+++ b/kernel/timeout.c
@@ -44,7 +44,7 @@ static struct _timeout *next(struct _timeout *t)
 	return n == NULL ? NULL : CONTAINER_OF(n, struct _timeout, node);
 }
 
-static void remove(struct _timeout *t)
+static void remove_timeout(struct _timeout *t)
 {
 	if (next(t) != NULL) {
 		next(t)->dticks += t->dticks;
@@ -95,7 +95,7 @@ int _abort_timeout(struct _timeout *to)
 
 	LOCKED(&timeout_lock) {
 		if (to->dticks != _INACTIVE) {
-			remove(to);
+			remove_timeout(to);
 			ret = 0;
 		}
 	}
@@ -140,7 +140,7 @@ void z_clock_announce(s32_t ticks)
 					announce_remaining -= t->dticks;
 					curr_tick += t->dticks;
 					t->dticks = 0;
-					remove(t);
+					remove_timeout(t);
 				} else {
 					t->dticks -= announce_remaining;
 					t = NULL;

